sources/core/objects/Interface.cpp

```diff
@@ -158,6 +168,54 @@ void HumanInterface::UpdateHealth()
         }
     }
 }
+void HumanInterface::UpdateEnviroment()
+{
+    if (IdPtr<Human> human = owner_)
+    {
+        if (IdPtr<CubeTile> t = human->GetOwner())
+        {
+            unsigned int oxygen = t->GetAtmosHolder()->GetGase(OXYGEN);
+            int temperature = t->GetAtmosHolder()->GetTemperature();
+            oxygen_.GetView()->SetState("oxy0");
+            if (oxygen < 1)
+            {
+                oxygen_.GetView()->SetState("oxy1");
+            }
+            if (temperature - 40 > 12)
+            {
+                temperature_.GetView()->SetState("temp4");
+            }
+            else if (temperature - 40 > 9)
```

Comment:
Health state now calculated the right way, the same should be made with temperature state.